### PR TITLE
[CDAP-14211] Batches runs count API in UI

### PR DESCRIPTION
**Pull request by ajainarayanan**
- Modifies the way UI uses runs endoint for each pipeline.
  - Queries runs endpoint with a limit of 1 only to the pipelines visible in UI 
  - Queries for next runtime endpoint to batch pipelines that are currently visible in UI
- Modifies the way UI gets runs count. Now uses the batch API added in backend.

**Note:**
 ~~- Since now UI only fetches the metadata (runs) for pipelines that are ONLY visible in UI right now the sorting is not possible. Users won't be able to sort by any field as UI doesn't have the entire data to sort. @bdmogal to update on this.~~
UPDATE:
  Decided to have a batch endpoint that gives latest runs for a list of pipelines. This will allow us to sort the pipelines by start time. However we still cannot sort by runs as the batch endpoint is fetched only for the currently visible pipelines.

~Adding on-hold for the corresponding backend PR: https://github.com/caskdata/cdap/pull/10560 to be merged.~

https://github.com/caskdata/cdap/pull/10653 & https://github.com/caskdata/cdap/pull/10560 merged. Removing on-hold

JIRA: https://issues.cask.co/browse/CDAP-14211
Build: https://builds.cask.co/browse/CDAP-UDUT80

### CODE REVIEW

**Comment by ajainarayanan** [comment]
Reference for on-hold [CDAP-14211](https://issues.cask.co/browse/CDAP-14211?focusedCommentId=33670&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-33670). 

**TL;DR** - Waiting for backend API for getting latest runs for list of pipelines as one batch call.

**Comment by ajainarayanan** [comment]
Backend PR: https://github.com/caskdata/cdap/pull/10653
Updated the UI based on backend batch runs endpoint api integration.

File: cdap-ui/app/hydrator/controllers/list-ctrl.js [hunk 1]
```diff
@@ -27,23 +33,20 @@ angular.module(PKG.name + '.feature.hydrator')
...
         Array.isArray(vm.pipelineList) &&
         Math.floor(vm.pipelineList.length) >= pageNumber
       );
     };
 
-    const setCurrentPage = () => {
+    vm.setCurrentPage = () => {
       let pageNumber = parseInt($stateParams.page, 10);
-      if (pageNumber&& checkForValidPage(pageNumber)) {
+      if (pageNumber&& vm.checkForValidPage(pageNumber)) {
```
Comment (elfenheart) [inline cdap-ui/app/hydrator/controllers/list-ctrl.js]: nit: space before `&&`

File: cdap-ui/app/hydrator/controllers/list-ctrl.js [hunk 2]
```diff
@@ -138,89 +139,139 @@ angular.module(PKG.name + '.feature.hydrator')
...
-        params.workerId = 'ETLWorker';
-
-        realtime.push({
-          appId: app.name,
-          programType: 'Worker',
-          programId: 'ETLWorker'
-        });
+    vm.getProgramId = (app) => {
+      if (app.artifact.name === GLOBALS.etlDataPipeline) {
+        return app.artifact.name === GLOBALS.etlDataPipeline ? 'DataPipelineWorkflow' : 'ETLWorkflow';
```
Comment (elfenheart) [inline cdap-ui/app/hydrator/controllers/list-ctrl.js]: can we perhaps safely deprecate this check and just go with `DataPipelineWorkflow`?

Comment (ajainarayanan) [reply cdap-ui/app/hydrator/controllers/list-ctrl.js]: Hhmm we could. There are a lot of references still in pipelines UI that refer to `cdap-etl-batch` and `cdap-etl-realtime`.

Comment (ajainarayanan) [reply cdap-ui/app/hydrator/controllers/list-ctrl.js]: Created a JIRA https://issues.cask.co/browse/CDAP-14317 to track removal of older pipeline artifacts from UI.

**Review by elfenheart — APPROVED** [review]
👍 LGTM
